### PR TITLE
book-store-impl-pagination-swagger

### PR DESCRIPTION
**Pull request by SkillfulDev**

### CODE REVIEW

File: src/main/java/ua/chernonog/onlinebookstore/dto/request/CreateBookRequestDto.java [hunk 1]
```diff
@@ -17,7 +17,7 @@ public class CreateBookRequestDto {
     @NotEmpty(message = "Author cannot be empty")
     private String author;
     @NotEmpty(message = "ISBN cannot be empty")
-    @Pattern(regexp = "^(?:\\d{9}[\\dX]|97[89]\\d{10})$", message = "Invalid ISBN format")
+    @Pattern(regexp = "(?:978|979)?\\d{9}[\\dX]", message = "Invalid ISBN format")
     private String isbn;
```
Comment (ReamFOX) [inline src/main/java/ua/chernonog/onlinebookstore/dto/request/CreateBookRequestDto.java]: There is a ready-made validator based on the @ISBN annotation, but I think it's ok to use the simplified validation here

**Review by ReamFOX — APPROVED** [review]
LGTM. Just minor comment.

File: src/main/java/ua/chernonog/onlinebookstore/dto/request/CreateBookRequestDto.java (hunk 1 above)
Comment (andrii-hoienko) [reply src/main/java/ua/chernonog/onlinebookstore/dto/request/CreateBookRequestDto.java]: Please consider following annotation

File: src/main/java/ua/chernonog/onlinebookstore/service/BookServiceImpl.java [hunk 2]
```diff
@@ -57,9 +58,9 @@ public BookDto updateById(Long id, CreateBookRequestDto requestDto) {
     }
 
     @Override
-    public List<BookDto> search(BookSearchParametersDto searchParameters) {
+    public List<BookDto> search(BookSearchParametersDto searchParameters, Pageable pageable) {
         Specification<Book> bookSpecification = specificationBuilder.build(searchParameters);
-        return bookRepository.findAll(bookSpecification).stream()
+        return bookRepository.findAll(bookSpecification,pageable).stream()
```
Comment (andrii-hoienko) [inline src/main/java/ua/chernonog/onlinebookstore/service/BookServiceImpl.java]:
```suggestion
        return bookRepository.findAll(bookSpecification, pageable).stream()
```

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Very nice, left minor comments

**Pull request closed without merging**